### PR TITLE
Adding multi-arch support to the files-cache web server

### PR DESCRIPTION
**Pull request by bradmwilliams**
The multi-arch folks reached out about the release-controller not supporting their respective installer downloads when accessing the Release Status webpage.  This PR adds support for the files-cache to pull the installers from their architecture specific pull specs.

### CODE REVIEW

File: cmd/release-controller/http_helper.go [hunk 1]
```diff
@@ -702,7 +702,11 @@ func renderInstallInstructions(w io.Writer, mirror *imagev1.ImageStream, tag *im
 		fmt.Fprintf(w, `<p>Download installer and client with:<pre class="ml-4">oc adm release extract --tools %s</pre>`, template.HTMLEscapeString(tagPull))
 		return
 	}
-	fmt.Fprintf(w, `<p><a href="%s">Download the installer</a> for your operating system or run <pre class="ml-4">oc adm release extract --tools %s</pre>`, template.HTMLEscapeString(fmt.Sprintf("https://%s/%s", artifactsHost, tag.Name)), template.HTMLEscapeString(tagPull))
+	pieces := strings.Split(tagPull, ":")
```
Comment (smarterclayton) [inline cmd/release-controller/http_helper.go]: This isn't safe, you can't assume a pull spec has a ":" (it could be a digest).  You'll need to parse this.

File: cmd/release-controller/info.go [hunk 2]
```diff
@@ -417,84 +417,131 @@ cp /tmp/pull-secret/* /tmp/.docker/ || true
 oc registry login
 
 cat <<END >>/tmp/serve.py
-import re, os, subprocess, time, threading, socket, SocketServer, BaseHTTPServer, SimpleHTTPServer
+import re
```
Comment (smarterclayton) [inline cmd/release-controller/info.go]: completely refactoring this makes this much harder to review.  If you want to restructure the code, do it in a separate commit before making substantive changes.

File: cmd/release-controller/info.go [hunk 3]
```diff
@@ -417,84 +417,131 @@ cp /tmp/pull-secret/* /tmp/.docker/ || true
...
+            except OSError:
+                pass
+
+            with open(os.path.join(name, "DOWNLOADING.md"), "w") as outfile:
+                outfile.write("Downloading %s" % name)
+
+            try:
+                self._present_default_content(name)
+
+                namespace = query_components['ns'][0].strip("/") if 'ns' in query_components else 'ocp'
```
Comment (jupierce) [inline cmd/release-controller/info.go]: Can we guard against probing of unrelated namespaces. 
```
if 'ns' not in query_components:
   namespace = {first release controller in release controller args}
else:
   namespace = query_components['ns'][0].strip("/")
   if namespace not in [{list of release namespaces associated with this release controller instance}]:
      return an error
```

File: cmd/release-controller/info.go [hunk 4]
```diff
@@ -417,84 +417,131 @@ cp /tmp/pull-secret/* /tmp/.docker/ || true
...
+                imagestream = query_components['is'][0].strip("/") if 'is' in query_components else 'release'
```
Comment (jupierce) [inline cmd/release-controller/info.go]: Can we regex this to make sure it's clean?

File: cmd/release-controller/http_helper.go (hunk 1 above)
Comment (bradmwilliams) [reply cmd/release-controller/http_helper.go]: @smarterclayton If a digest is passed in, do we want to process it further, or just post the "Download the installer with:..." message?

File: cmd/release-controller/info.go (hunk 2 above)
Comment (bradmwilliams) [reply cmd/release-controller/info.go]: My bad.  I've broken the changes down into more meaningful chunks starting with the refactoring piece and then modifications.
